### PR TITLE
TEC-950: Feature/sticky sharebar

### PR DESCRIPTION
**Pull request by dmolin**
Implementing stickyness for the sharebar elements area.
There was no ideal way to do this, so I opted for the most sensible tradeoff.

A pure CSS solution was implemented first, but it had the limit that the height of the sticky container had to be hardcoded without being able to relate it to the current article height. it was "OK" but not always ideal.

Since the requirement calls for having the share bar always in view up to the end of the article, a JS solution needed to be implemented. "where" to code it has been the main issue:

The problem was that the 'pulling out' of the share block area from the article happens outside of the component (in fe-blogs->BlogPostLayout, as a pure CSS operation). Controlling the share block stickiness via code over there would've been a violation of encapsulation (since the share block is owned by the blog-post component). Hence, I decided to add the relevant code within the component-blog-post itself.

The main idea is that whenever the aside shareblock, for whatever reason, gets pulled out from its own container (the article), then some special logic is applied, that simply resize its immediate container so that its bottom is in line with the article. This behaviour gets applied only when the 'aside' area is not within the horizontal boundaries of the article. 
Since, the pulling out (and the absolute positioning) is performed outside of the component, the only way to test its behaviour (apart from the unit tests) is to link the component and check it locally in the article page...
Note: this component relies on the updated npm module component-stickyfill (that was upgraded to babel 6)

### CODE REVIEW

**Comment by dmolin** [comment]
ah yes, right. At the moment the build is failing on Travis, since this component requires the @economist/component-stickyfill component to be babel6 compliant...

File: package.json [hunk 1]
```diff
@@ -112,11 +112,13 @@
     "@economist/component-sharebar": "^3.2.0",
     "@economist/component-typography": "^3.1.4",
     "classnames": "^2.2.3",
+    "lodash": "^4.13.1",
```
Comment (gcedo) [inline package.json]: You could install just the [`lodash.throttle`](https://www.npmjs.com/package/lodash.throttle) package, as to reduce build size

File: src/index.js [hunk 2]
```diff
@@ -176,6 +210,48 @@ export default class BlogPost extends React.Component {
...
+    // this rule applies only when the element gets moved out of the container horizontal space
+    if (elementRect.left >= containerRect.left &&
+        elementRect.right <= containerRect.right) {
+      return true;
+    }
+    return false;
+  }
+
+  checkAsideableContainerHeight() {
+    const asideContainerEl = this.getDOMElementFromRef(this.refs.asideable);
```
Comment (gcedo) [inline src/index.js]: Nitpick, I think we could avoid the function here and just call `ReactDOM.find`.

**Comment by umbobabo** [comment]
LGTM. @gcedo can merge if he is happy.

**Pull request merged**